EIPS/eip-4762.md

```diff
@@ -39,47 +39,43 @@ We define access events as follows. When an access event takes place, the access
 
 #### Access events for account headers
 
-When a non-precompile address is the target of a `CALL`, `CALLCODE`, `DELEGATECALL`, `SELFDESTRUCT`, `EXTCODESIZE`, or `EXTCODECOPY` opcode, or is the target address of a contract creation whose initcode starts execution, process these access events:
+When:
 
-```
-(address, 0, VERSION_LEAF_KEY)
-(address, 0, CODE_SIZE_LEAF_KEY)
-```
+ 1. a non-precompile address is the target of a `CALL`, `CALLCODE`, `DELEGATECALL`, `STATICALL`, `AUTHCALL`, `SELFDESTRUCT`, `EXTCODESIZE`, or `EXTCODECOPY` opcode, 
+ 2. a non-precompile address is the target address of a contract creation whose initcode starts execution,
+ 3. any address is the target of the `BALANCE` opcode
+ 4. a _deployed_ contract calls `CODECOPY`
 
-If a call is value-bearing (ie. it transfers nonzero wei), whether or not the callee is a precompile, process these two access events:
+process this access events:
 
 ```
-(caller_address, 0, BALANCE_LEAF_KEY)
-(callee_address, 0, BALANCE_LEAF_KEY)
+(address, 0, BASIC_DATA_LEAF_KEY)
 ```
 
-When a contract is created, process these access events:
+If a `*CALL` or `SELFDESTRUCT` is value-bearing (ie. it transfers nonzero wei), whether or not the callee is a precompile, process these additional two access events:
 
 ```
-(contract_address, 0, VERSION_LEAF_KEY)
-(contract_address, 0, NONCE_LEAF_KEY)
-(contract_address, 0, BALANCE_LEAF_KEY)
-(contract_address, 0, CODE_KECCAK_LEAF_KEY)
-(contract_address, 0, CODE_SIZE_LEAF_KEY)
+(caller_address, 0, BASIC_DATA_LEAF_KEY)
+(callee_address, 0, BASIC_DATA_LEAF_KEY)
 ```
 
-If the `BALANCE` opcode is called targeting some address, process this access event:
-
+When calling `EXTCODEHASH`, process the access event:
 ```
-(address, 0, BALANCE_LEAF_KEY)
+(address, 0, CODE_HASH_LEAF_KEY)
 ```
 
-If the `SELFDESTRUCT` opcode is called by some caller_address targeting some target_address (regardless of whether it’s value-bearing or not), process access events of the form:
+When a contract is created, process these access events:
 
 ```
-(caller_address, 0, BALANCE_LEAF_KEY)
-(target_address, 0, BALANCE_LEAF_KEY)
+(contract_address, 0, BASIC_DATA_LEAF_KEY)
+(contract_address, 0, CODEHASH_LEAF_KEY)
 ```
 
-If the `EXTCODEHASH` opcode is called targeting some address, process an access event of the form:
+If the `SELFDESTRUCT` opcode is called by some `caller_address` targeting some `target_address` (regardless of whether it’s value-bearing or not), process access events of the form:
 
 ```
-(address, 0, CODEHASH_LEAF_KEY)
+(caller_address, 0, BASIC_DATA_LEAF_KEY)
+(target_address, 0, BASIC_DATA_LEAF_KEY)
```

Comment:
So, isn't this L74-L78 rule more general than [this one](https://github.com/ethereum/EIPs/pull/8550/files#diff-cd2aaa222faeb6081a41605386a509b3c4839285cd15f1b0ebe009318707914eR55) for `SELFDESTRUCT`?

If that's the case, we could remove `SELFDESTRUCT` mentioned in that linked rule, and only leave this L74-L78?

Reply:
I'm starting to feel if wouldn't be easier for everyone to simply list rules for each independent instruction. Like `*CALL`, `SELFDESTRUCT`, etc, instead of trying to coalesce overlaps and re-clarify later with further rules some accessed extensions.

Yes, separating rules by instructions might have some redundancy but for implementors and testing it will be entirely clear what to do for each instruction. Now you have to read the whole document to see if in an extra scenario the same instruction you're interested in has another rule.

Reply:
yes, i kind of agree with this. Listing every opCode seperately will be a better idea. 
Right now its much easier to miss something.

Reply:
I agree, it's hard to follow specs like that, I also understand the idea of ​​having a concise EIP